python/dgl/graphbolt/impl/ondisk_dataset.py

```diff
@@ -16,13 +32,183 @@
+                )
+                g.edata[graph_feature["name"]] = edge_data
+
+    # 4. Convert the DGLGraph to a CSCSamplingGraph.
+    csc_sampling_graph = from_dglgraph(g)
+
+    # 5. Save the CSCSamplingGraph and modify the output_config.
+    output_config["graph_topology"] = {}
+    output_config["graph_topology"]["type"] = "CSCSamplingGraph"
+    output_config["graph_topology"]["path"] = (
+        processed_dir_prefix / "csc_sampling_graph.tar"
+    )
+
+    save_csc_sampling_graph(
+        csc_sampling_graph,
+        dataset_path / output_config["graph_topology"]["path"],
+    )
+    del output_config["graph"]
+
+    # 6. Load the node/edge features and do necessary conversion.
+    if input_config.get("feature_data", None):
+        for (feature, out_feature) in zip(
+            input_config["feature_data"], output_config["feature_data"]
+        ):
+            # Always save the feature in numpy format.
+            out_feature["format"] = "numpy"
+            out_feature["path"] = processed_dir_prefix / feature[
+                "path"
+            ].replace("pt", "npy")
+
+            if feature["format"] == "torch":
+                data = read_data(
+                    dataset_path / feature["path"],
+                    feature["format"],
+                    in_memory=feature["in_memory"],
+                )
+                save_data(
+                    data,
+                    dataset_path / out_feature["path"],
+                    out_feature["format"],
+                )
+            elif feature["format"] == "numpy":
+                # If the original format is numpy, just copy the file.
+                os.makedirs(
+                    dataset_path / os.path.dirname(out_feature["path"]),
+                    exist_ok=True,
+                )
+                shutil.copyfile(
+                    dataset_path / feature["path"],
+                    dataset_path / out_feature["path"],
+                )
+            else:
+                raise ValueError(
+                    f"Unsupported feature format {feature['format']}"
+                )
+
+    # 7. Save the train/val/test split according to the output_config.
+    for set_name in ["train_sets", "validation_sets", "test_sets"]:
+        if set_name not in input_config:
+            continue
+        for intput_set_split, output_set_split in zip(
+            input_config[set_name], output_config[set_name]
+        ):
+            for input_set_per_type, output_set_per_type in zip(
+                intput_set_split, output_set_split
+            ):
+                # Always save the feature in numpy format.
+                output_set_per_type["format"] = "numpy"
+                output_set_per_type[
+                    "path"
+                ] = processed_dir_prefix / input_set_per_type["path"].replace(
+                    "pt", "npy"
+                )
+                if input_set_per_type["format"] != "numpy":
```

Comment:
Does the following conversion supports all kinds of format or just torch?

Reply:
Currently we only support `torch` and `numpy` formats.

Reply:
Let's follow the pattern in # 6. It is less error prone in the future when we introduce a new format.

Reply:
Is it resolved?